### PR TITLE
Adds frr project

### PR DESCRIPTION
**Pull request by catenacyber**
@qlyoung would you be interested in continuous fuzzing for FRR ?
This enables with oss-fuzz with the fuzz targets and corpuses already present and documented

### CODE REVIEW

File: projects/frr/build.sh
Comment (asraa) [inline projects/frr/build.sh:45]: what about vrrpd?

**Review by asraa — COMMENTED** [review]
Thanks! Couple of comments.
Generally approve of the PR, it is widely used, has active releases and activity.

File: projects/frr/project.yaml
Comment (qlyoung) [inline projects/frr/project.yaml:8]: Earlier on you've built FRR for libfuzzer, which if I recall correctly, excludes the AFL entrypoints (libfuzzer provides its own `main` and the AFL entrypoints are in our `main`, so that's compiled out when libfuzzer is enabled).

**Review by qlyoung — COMMENTED** [review]
So while I support getting this in and I'm vaguely familiar with oss-fuzz, I'm not familiar with the technicals on how this is going to work.

The code I follow; in a docker image, build and install our dependent libraries, build and install FRR, mark the appropriate fuzzers.

My questions are:
- Are the results oss-fuzz generates sent only to the primary contact? Is there any expectation of turnaround time or is this like e.g. coverity where a scan runs once in a while, we have access to reports, etc.
- Say I break the `fuzz` branch (e.g. fails to compile): how is oss-fuzz going to react?
- Are there mechanisms to suppress false positives, if they should arise?
- Who has "control" over the project being part of oss-fuzz or not? It seems to me that should rest with the governing body for FRR (of which I am a member). I want to make sure that's clear and I don't see a mechanism that gives us that authority

To @catenacyber specifically, I'd like to know what your intention is by putting yourself on the CC list. I assume that means you get security reports for our project and given your email address belongs to a security company, I think we (FRR) might have some objections to that, as much as I appreciate you doing the work here.

**Comment by asraa** [comment]
Thanks for the reply! I'll try to answer my best:

> Are the results oss-fuzz generates sent only to the primary contact? Is there any expectation of turnaround time or is this like e.g. coverity where a scan runs once in a while, we have access to reports, etc.

The reports will also be available to the auto_ccs. The primary contact is just the primary (in case there's some PR changing the config here we will request your LGTM). Yeah, what happens is the fuzzers will be rebuilt every day, and oss-fuzz will automatically de-dupe, bisect, create bug reports with stack traces, and notify the auto_ccs/contact of the report. If a fix lands upstream, then the bug is automatically closed. You'll have access to the reports continuously.

> Say I break the fuzz branch (e.g. fails to compile): how is oss-fuzz going to react?

A build breakage bug will be filed. The build logs are always available at https://oss-fuzz-build-logs.storage.googleapis.com/index.html

> Are there mechanisms to suppress false positives, if they should arise?

Bugs that are unreproducible are tagged and eventually oss-fuzz will close those out if they don't reproduce anymore. For true false positives you can mark them as WontFix

> Who has "control" over the project being part of oss-fuzz or not? It seems to me that should rest with the governing body for FRR (of which I am a member). I want to make sure that's clear and I don't see a mechanism that gives us that authority

I don't think I 100% understand this question but at any point you can remove the project by removing its configuration in OSS-Fuzz, you can deny this PR to add the project (and we won't merge it), you can request a different primary contact who we'll ask for LGTMs for any changes, etc.

File: projects/frr/build.sh
Comment (catenacyber) [reply projects/frr/build.sh:52]: I do not understand the question

File: projects/frr/project.yaml
Comment (catenacyber) [reply projects/frr/project.yaml:8]: The way oss-fuzz works is to have the same libFuzzer entry points and run them with afl with a specific afl driver.
Do I get it correct @asraa ?

**Comment by catenacyber** [comment]
>  I'd like to know what your intention is by putting yourself on the CC list. I assume that means you get security reports for our project and given your email address belongs to a security company, I think we (FRR) might have some objections to that, as much as I appreciate you doing the work here.

@qlyoung I just removed myself.

**Comment by catenacyber** [comment]
@qlyoung 

It looks like the build is broken now :
```
make[1]: Entering directory '/src/frr'
  CC       bgpd/bgp_addpath.o
In file included from bgpd/bgp_addpath.c:24:
In file included from bgpd/bgp_addpath.h:24:
In file included from ./lib/zebra.h:35:
./lib/compiler.h:155:12: error: expected parameter declarator
CPP_NOTICE("time to remove this CONFDATE block")
           ^
./lib/compiler.h:155:12: error: expected ')'
```

Do you know about these ?
They seem intentional messages

**Comment by qlyoung** [comment]
Both of you answered my questions sufficiently, thank you. I think at this point I should probably consult with the rest of the FRR team and make sure we have a good consensus on our inclusion in oss-fuzz. Again, thanks a lot for doing the work here. I'll update this PR when we can proceed, it shouldn't be very long.

@catenacyber when we plan to deprecate things in the codebase, we insert a macro that will cause builds to fail after a certain date; typically this date is 1 year in the future. So we declare something deprecated, add the macro, when the build breaks we remove the macro and the deprecated code. Happens a few times a year usually.

Since the branch containing fuzzing targets (`fuzz`) is manually maintained by me, I'll need to rebase the branch to fix this. This will probably happen a few times a year.

**Comment by inferno-chromium** [comment]
Thanks @qlyoung, leave a comment here and then we can merge.

**Comment by qlyoung** [comment]
Build issue ought to be fixed now.

Spoke with our project body - we should be good to merge this one. Thanks!

**Review by qlyoung — APPROVED** [review]

**Comment by catenacyber** [comment]
Just force-pushed on rebased version to get CI checking the build

**Comment by qlyoung** [comment]
Looks like it's missing libpcre?

**Comment by catenacyber** [comment]
> Looks like it's missing libpcre?

We need to use the static version instead of the shared one of libpcre...
Or hack some rpath stuff...
I will take a look

File: projects/frr/project.yaml
Comment (asraa) [reply projects/frr/project.yaml:3]: ping on this comment @qlyoung 

with no auto_ccs no one has detailed access

Comment (qlyoung) [reply projects/frr/project.yaml:3]: Roger, you can add qlyoung |the usual symbol| qlyoung.net

Comment (catenacyber) [reply projects/frr/project.yaml:3]: Done ;)

**Review by asraa — APPROVED** [review]
LGTM, thanks! Pending CI

**Comment by qlyoung** [comment]
What is needed here?

**Comment by inferno-chromium** [comment]
> What is needed here?

someone forgot to merge, done now.

**Pull request merged**
